### PR TITLE
Add: plugin for finding excess space before dot in some tags

### PR DESCRIPTION
**Pull request by NiklasHargarter**
## What
Adds a plugin to check script_tags that use full sentences in their value for excess white spaces before sentence terminating dots.
fix method is implemented to automatically remove the preceding excess white spaces. 

21.04 and 22.04 are ignored because they are generated.
Other entries in the ignore list use whitespace in various ways that have excess whitespace before dots, but on purpose, as used for beginning entries in an enumeration.

## References
Jira VTOPS-229

## Checklist

- [x] Tests

currently finds 121 occurrences in VT repo,

### CODE REVIEW

File: tests/plugins/test_spaces_before_dots.py
Comment (cfi-gb) [inline tests/plugins/test_spaces_before_dots.py:16]:
```suggestion
              script_tag(name:"insight", value:"foo bar ...");
              script_tag(name:"solution", value:"meh.");
```

Might be valid so i would not report these

File: tests/plugins/test_spaces_before_dots.py
Comment (cfi-gb) [inline tests/plugins/test_spaces_before_dots.py:47]: As shown in VTOPS-229 i have also seen something like this "live" / happening which should also be reported:

```
  script_tag(name:"affected", value:"Adobe Reader 2017 before 2017.011.30166
 .");
```

File: troubadix/plugins/spaces_before_dots.py
Comment (NiklasHargarter) [reply troubadix/plugins/spaces_before_dots.py:38]: there are many valid occurences of spaces before dots liike with .net and other words that start with a '.'

Comment (NiklasHargarter) [reply troubadix/plugins/spaces_before_dots.py:38]: if going with regex and finding:
1. dots enclosed in whitespace "foo . bar"
2. whitespace before end of string: "foo ." "foo\n"
I could probably use '\s\.\s|\s\.$'

File: tests/plugins/test_spaces_before_dots.py
Comment (NiklasHargarter) [reply tests/plugins/test_spaces_before_dots.py]: tripple dots will not trigger a report. But included as a test in 9a8f9bdcf577ce0a5fc421742bfc464adbcfc1e4 to make sure.

File: tests/plugins/test_spaces_before_dots.py
Comment (NiklasHargarter) [reply tests/plugins/test_spaces_before_dots.py]: Completely overlooked that newline should be included. Now it covers all white space characters and not only spaces. Done in 9a8f9bdcf577ce0a5fc421742bfc464adbcfc1e4

File: troubadix/plugins/spaces_before_dots.py
Comment (NiklasHargarter) [reply troubadix/plugins/spaces_before_dots.py]: Changed to the regex above in 9a8f9bdcf577ce0a5fc421742bfc464adbcfc1e4

Comment (NiklasHargarter) [reply troubadix/plugins/spaces_before_dots.py]: done in 9a8f9bdcf577ce0a5fc421742bfc464adbcfc1e4

**Review by cfi-gb — COMMENTED** [review]
The existing test cases LGTM, thanks a lot for taking care :+1: 

I would leave the final review, approve and merge for one of the @greenbone/troubadix-maintainers .

Also not sure if it would make sense to split the `test_fail` case into single ones for each case but would also leave this up to the @greenbone/troubadix-maintainers

File: troubadix/plugins/spaces_before_dots.py
Comment (NiklasHargarter) [reply troubadix/plugins/spaces_before_dots.py]: The '+" can also be left out. 
Changed to the short regex in 1237ff6ae295dbb0c88a388d8d6c288c596b52f7

File: tests/plugins/test_spaces_before_dots.py
Comment (cfi-gb) [inline tests/plugins/test_spaces_before_dots.py:71]:
```suggestion
                  script_tag(name:"insight", value:"Foo .NET Bar .");
```

File: tests/plugins/test_spaces_before_dots.py
Comment (cfi-gb) [inline tests/plugins/test_spaces_before_dots.py:81]:
```suggestion
                  script_tag(name:"insight", value:"Foo .NET Bar.");
```

For the previous suggestion

File: tests/plugins/test_spaces_before_dots.py
Comment (NiklasHargarter) [reply tests/plugins/test_spaces_before_dots.py]: fixed in f7c8088f9b6cf5074e02fca745abc4eb5dd04f9c

File: tests/plugins/test_spaces_before_dots.py
Comment (NiklasHargarter) [reply tests/plugins/test_spaces_before_dots.py]: fixed in f7c8088f9b6cf5074e02fca745abc4eb5dd04f9c

File: troubadix/plugins/spaces_before_dots.py
Comment (NiklasHargarter) [reply troubadix/plugins/spaces_before_dots.py]: fixed in f7c8088f9b6cf5074e02fca745abc4eb5dd04f9c

**Comment by NiklasHargarter** [comment]
Wait for confirmation that 22.04 should also be completely excluded.

File: troubadix/plugins/spaces_before_dots.py
Comment (n-thumann) [inline troubadix/plugins/spaces_before_dots.py:110]:
```suggestion
        self.matches.sort(reverse=True)
```
`whitespace_pos` is unique, so that there can only ever be one tuple with the same, right?
In that case, we could drop the `key`, because Python will first compare by the first tuple element and only then consider the second (which won't happen, because the first element is unique anyways).
If there are two elements with the same `whitespace_pos`, itemgetter will (probably) keep the original order, instead of then taking the second tuple element into account.

**Review by n-thumann — DISMISSED** [review]

File: troubadix/plugins/spaces_before_dots.py
Comment (NiklasHargarter) [reply troubadix/plugins/spaces_before_dots.py]: done in 0ebb27b978aa83a14db59d3c9552f0834879b433

File: troubadix/plugins/spaces_before_dots.py
Comment (NiklasHargarter) [reply troubadix/plugins/spaces_before_dots.py]: done in 0ebb27b978aa83a14db59d3c9552f0834879b433

**Pull request merged**
